src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```diff
@@ -0,0 +1,269 @@
+            return storage.GetInteroperable<TAccount>().Balance;
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.Array, CallFlags.AllowStates, ParameterTypes = new[] { ContractParameterType.Hash160 }, ParameterNames = new[] { "owner" })]
+        public StackItem TokensOf(ApplicationEngine engine, Array args)
+        {
+            UInt160 owner = new UInt160(args[0].GetSpan());
+            return new InteropInterface(TokensOf(engine.Snapshot, owner));
+        }
+
+        public virtual IEnumerator TokensOf(StoreView snapshot, UInt160 owner)
+        {
+            return new CollectionWrapper(snapshot.Storages.Find(CreateStorageKey(Prefix_OwnerToTokenId, owner).ToArray()).Select(p =>
+            {
+                UInt160 innerKey = new UInt160(p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray());
+                TToken token = snapshot.Storages.TryGet(CreateTokenKey(innerKey)).GetInteroperable<TToken>();
+                return (StackItem)token.Id;
+            }).GetEnumerator());
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.InteropInterface, CallFlags.None, ParameterTypes = new[] { ContractParameterType.ByteArray }, ParameterNames = new[] { "tokenId" })]
+        public StackItem OwnerOf(ApplicationEngine engine, Array args)
+        {
+            byte[] tokenId = args[0].GetSpan().ToArray();
+            if (tokenId.Length > MaxTokenIdLength) return false;
+            return new InteropInterface(OwnerOf(engine.Snapshot, tokenId));
+        }
+
+        public virtual IEnumerator OwnerOf(StoreView snapshot, byte[] tokenId)
+        {
+            UInt160 innerKey = GetInnerKey(tokenId);
+            return new CollectionWrapper(snapshot.Storages.Find(CreateStorageKey(Prefix_TokenIdToOwner, innerKey).ToArray()).Select(p =>
+            {
+                var owner = p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray();
+                return (StackItem)owner;
+            }).GetEnumerator());
+        }
+
+        [ContractMethod(0_08000000, ContractParameterType.Boolean, CallFlags.AllowModifyStates, ParameterTypes = new[] { ContractParameterType.Hash160, ContractParameterType.ByteArray }, ParameterNames = new[] { "to", "tokenId" })]
+        public virtual StackItem Transfer(ApplicationEngine engine, Array args)
+        {
+            UInt160 to = new UInt160(args[0].GetSpan());
+            byte[] tokenId = args[1].GetSpan().ToArray();
+            if (tokenId.Length > MaxTokenIdLength) return false;
+            return Transfer(engine, to, tokenId);
+        }
+
+        public virtual bool Transfer(ApplicationEngine engine, UInt160 to, byte[] tokenId)
+        {
+            IEnumerator enumerator = OwnerOf(engine.Snapshot, tokenId);
+            if (!enumerator.Next()) return false;
+            UInt160 owner = new UInt160(enumerator.Value().GetSpan().ToArray());
+            if (!engine.CheckWitnessInternal(owner)) return false;
```

Comment:
If the owner it's the second in the enumerator?

Reply:
It has only one owner, when `Decimals = 0`.

Reply:
This will not happen, because in this contract NFT is restricted to be indivisible, so there is no second owner.

Reply:
Then why the enumerator?

Reply:
We need `IEnumerator ` for  `tokensOf` methods of nep11 .
```cs
public static enumerator tokensOf(byte[] owner)
```

Reply:
Maybe we can use `UInt160` as the return type of the internal `tokensOf`, it's ok for me.